### PR TITLE
Strip quotes from maintainer names in the announcement

### LINKED ISSUE #271: Some names in maintainers list in sync output are quoted
I noticed that the names in the list of maintainers are now quoted (as of a few months ago?). For example, see https://discourse.openrobotics.org/t/new-packages-for-humble-hawksbill-2026-08-07/57228

<details>

<summary>Click to expand</summary>

```
Thanks to all ROS maintainers who make packages available to the ROS community. The above list of packages was made possible by the work of the following maintainers: 

 * "Adam Dabrowski"
 * "Aditya Pande"
 * "Alejandro Hernanadez"
 * "Alexander Mock"
 * "Ali Pahlevani"
 * "Aron Svastits"
 * "Audrow Nash"
 * "Autoware"
 * "Bence Magyar"
 * "Berkay Karaman"
 * "Chittaranjan Swaminathan"
 * "Chris Lalancette"
 * "Daniel Stonier"
 * "Dave Coleman"
 * "David Wong"
 * "Davide Costa"
 * "Davide Faconti"
 * "Ethan Gao"
 * "Felix Exner"
 * "Foxglove"
 * "Fumiya Watanabe"
 * "G.A. vd. Hoorn"
 * "Geoffrey Biggs"
 * "Gergely Kovacs"
 * "Hung Pham"
 * "Isaac I.Y. Saito"
 * "Ivan Paunovic"
 * "Jacob Perron"
 * "Jean-Pierre Busch"
 * "Jian Kang"
 * "Jose Luis Blanco-Claraco"
 * "Jose-Luis Blanco-Claraco"
 * "Junya Sasaki"
 * "Justin Carpentier"
 * "Kosuke Takeuchi"
 * "Kyoichi Sugahara"
 * "Lennart Reiher"
 * "Luis Camero"
 * "M. Fatih Cırıt"
 * "Mamoru Sobue"
 * "Manan Kharwar"
 * "Masahiro Sakamoto"
 * "Max Schmeller"
 * "Maxime CLEMENT"
 * "Maxime Clement"
 * "Meher Malladi"
 * "Mete Fatih Cırıt"
 * "Miguel Ángel González Santamarta"
 * "Mike Lautman"
 * "Mitsuhiro Sakamoto"
 * "MoveIt Release Team"
 * "Piotr Jaroszek"
 * "Raul Sanchez Mateos"
 * "Roni Kreinin"
 * "Ryohsuke Mitsudome"
 * "SBG Systems"
 * "Sai Kishor Kothakota"
 * "Satoshi OTA"
 * "Satoshi Ota"
 * "Shigeru Wakida"
 * "Shunsuke Kimura"
 * "Southwest Research Institute"
 * "Taiki Tanaka"
 * "Takagi, Isamu"
 * "Takahiro Ishikawa-Aso"
 * "Takamasa Horibe"
 * "Takayuki Murooka"
 * "Takumi Odashima"
 * "Team Spatzenhirn"
 * "Temkei Kem"
 * "Ubuntu Robotics"
 * "Wolfgang Hönig"
 * "Xingang Liu"
 * "Yadunund"
 * "Yamato Ando"
 * "Yoshi Ri"
 * "Yuki Takagi"
 * "Yukihiro Saito"
 * "Yukinari Hisaki"
 * "Zulfaqar Azmi"
 * "amc-nu"
 * "bmagyar"
 * "frozenreboot"
 * "lazytatzv"
 * "li9i"
 * "mitsudome-r"
 * "xingang liu"
 * Fabian Koenig
 * LibRealSense ROS Team
 * Maik Knof
 * Miguel Ángel González Santamarta
 * Stefan Laible
 * Tibor Dome
```

</details>

### PR DESCRIPTION
**Pull request by christophebedard**
Fixes #271

bloom quotes the name in debian/control since 0.14.3: https://github.com/ros-infrastructure/bloom/pull/770. This looks weird and also results in a maintainer being listed twice when they released packages with both an older and a newer bloom.

I added tests to cover this, which include corner cases we don't really encounter much, including `Maintainer:` entries with multiple comma-separated names (because you can just define separate `<maintainer/>`). However, it could happen, or the maintainer name might be the name of a company and include a comma.

I kept these tests to document the behaviour in that case.

This targets the `refactor` branch, because that's what all `*rel_sync-packages-to-main` jobs seem to use, not `master`.

---

I used AI (Claude Opus 5) to find this issue and make the initial fix, which I then reviewed and re-worked.

### CODE REVIEW

File: src/reprepro_updater/diff_repos.py [hunk 1]
```diff
@@ -27,7 +27,17 @@ def convert_tuples_list_to_dict(tuples_list):
...
+    """Extract the maintainer name from a debian Maintainer field."""
+    name = re.sub("(.*)<.*>", "\\1", maintainer).strip()
+    # bloom wraps the name in double quotes to keep the field RFC 5322 compliant, see:
+    # https://github.com/ros-infrastructure/bloom/pull/770. Unwrap the name.
+    # Only unwrap a pair of quotes around the whole name: a field holding more than one mailbox also
+    # starts and ends with a quote, and stripping those would corrupt it rather than leave it merely
+    # ugly.
+    unwrapped = re.match('^"([^"]*)"$', name)
+    if unwrapped:
+        name = unwrapped.group(1)
```
Comment (cottsay) [inline src/reprepro_updater/diff_repos.py]: This doesn't handle mid-string quotes. I'm not sure what the RFC has to say about that, but would it be easier to just check if the first and last characters are quotes and chop them off with a substring instead?


```suggestion
    if name.startswith('"') and name.endswith('"'):
        name = name[1:-1]
```

Comment (christophebedard) [reply src/reprepro_updater/diff_repos.py]: mid-string quotes sounds like a reasonable use-case, done

**Review by cottsay — APPROVED** [review]
